caravel/assets/javascripts/modules/caravel.js

```diff
@@ -76,16 +77,13 @@ const px = function () {
       container,
       containerId,
       selector,
-      querystring(params) {
-        const newParams = params || {};
+      querystring() {
         const parser = document.createElement('a');
         parser.href = data.json_endpoint;
-        if (dashboard !== undefined) {
-          let flts = '';
-          if (newParams.extraFilters !== false) {
-            flts = dashboard.effectiveExtraFilters(sliceId);
-            flts = encodeURIComponent(JSON.stringify(flts));
-          }
+        if (controller.type === 'dashboard') {
```

Comment:
Would you please describe the interface with a few comments. e.g. describe the case when each function is called.